metrics/pcp/pcp.go

```diff
@@ -0,0 +1,129 @@
+package pcp
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/performancecopilot/speed"
+)
+
+// Reporter encapsulates a speed client
+type Reporter struct {
+	c *speed.PCPClient
+}
+
+// NewReporter creates a new reporter instance
+func NewReporter(appname string) *Reporter {
+	c, err := speed.NewPCPClient(appname)
+	if err != nil {
+		panic(err)
+	}
+	return &Reporter{c}
+}
+
+// Start starts reporting currently registered metrics to the backend
+func (r *Reporter) Start() { r.c.MustStart() }
+
+// Stop stops reporting currently registered metrics to the backend
+func (r *Reporter) Stop() { r.c.MustStop() }
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+// Counter implements metrics.Counter via a single dimensional speed.Counter
+// for now, see https://github.com/performancecopilot/speed/issues/32
+type Counter struct {
+	c speed.Counter
+}
+
+// NewCounter creates a new Counter
+//
+// this requires a name parameter
+// and optionally takes a couple of string directions, which
+// are directly passed to speed
+func (r *Reporter) NewCounter(name string, desc ...string) *Counter {
+	c, err := speed.NewPCPCounter(0, name, desc...)
+	if err != nil {
+		panic(err)
+	}
+
+	r.c.MustRegister(c)
+	return &Counter{c}
+}
+
+// With is a no-op.
+func (c *Counter) With(labelValues ...string) metrics.Counter { return c }
+
+// Add implements Counter.
+// speed Counters only take int64
+// if it is important, instead use speed.SingletonMetric with DoubleType, CounterSemantics and OneUnit
+// but that will mean this will need a mutex, to be safe
```

Comment:
advising using metrics.Gauge for float64 additions, as counters only support int64, that is a design decision from the Java project speed itself is based on.